### PR TITLE
CB-10255. Ability to use DataBus CNAME instead of the sigma DataBus e…

### PR DESCRIPTION
**Pull request by oleewere**
…ndpoint.

details:
add new entitlement to let customers to use databus cname endpoint instead of the original one

the value would be: CDP_USE_DATABUS_CNAME_ENDPOINT

See detailed description in the commit message.

### CODE REVIEW

**Review by lnardai — APPROVED** [review]

File: common/src/main/java/com/sequenceiq/cloudbreak/telemetry/DataBusEndpointProvider.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataBusEndpointProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataBusEndpointProvider.class);
+
+    private static final String CNAME_ENDPOINT = "https://dbusapi.us-west-1.altus.cloudera.com";
```
Comment (horadla23) [inline common/src/main/java/com/sequenceiq/cloudbreak/telemetry/DataBusEndpointProvider.java:12]: Are we sure this URL won't change? If not, then maybe it is worth to make a new config property for this in application.yml

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/DiagnosticsTriggerService.java [hunk 2]
```diff
@@ -71,9 +81,12 @@ public FlowIdentifier startDiagnosticsCollection(BaseDiagnosticsCollectionReques
                 && StringUtils.isNotBlank(stack.getCluster().getBlueprint().getStackVersion())) {
             clusterVersion = stack.getCluster().getBlueprint().getStackVersion();
         }
+        String accountId = Crn.fromString(stack.getResourceCrn()).getAccountId();
+        boolean useDbusCnameEndpoint = entitlementService.useDataBusCNameEndpointEnabled(INTERNAL_ACTOR_CRN, accountId);
+        String databusEndpoint = dataBusEndpointProvider.getDataBusEndpoint(telemetry.getDatabusEndpoint(), useDbusCnameEndpoint);
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/DiagnosticsTriggerService.java:86]: This logic is very similar to the logic in TelemetryDecorator and in ClouderaManagerMgmtTelemetryService, might worth to make an own method for it somewhere

Comment (oleewere) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/DiagnosticsTriggerService.java:86]: common has no auth-connector dependency, first i tried to put these kind of code together in auth-connector but @lnardai did not want that (I understand) - so i cannot include Entitlementservice in DataBusEndpointProvider

File: common/src/main/java/com/sequenceiq/cloudbreak/telemetry/DataBusEndpointProvider.java (hunk 1 above)
Comment (oleewere) [reply common/src/main/java/com/sequenceiq/cloudbreak/telemetry/DataBusEndpointProvider.java:12]: does not seemed to be changed ever. I can do a follow-up change if that will be needed. although if it changes it should be changed everywhere so actually it sounds more like a constant still

Comment (horadla23) [reply common/src/main/java/com/sequenceiq/cloudbreak/telemetry/DataBusEndpointProvider.java:12]: ok, then it is fine for me

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/DiagnosticsTriggerService.java (hunk 2 above)
Comment (horadla23) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/DiagnosticsTriggerService.java:86]: ohh i see, makes sense

**Review by horadla23 — APPROVED** [review]

**Pull request merged**